segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java

```diff
@@ -396,6 +396,10 @@ public SegmentHandle openWrite(String streamSegmentName) throws StreamSegmentExc
                         } catch (FileNotFoundException e) {
                             //This happens when more than one host is trying to fence and only one of the host goes through.
                             //Retry the rename so that host with the highest epoch gets access.
+                            //In the worst case, the current owner of the segment will win this race after retries equal
```

Comment:
"after a number of attempts equal to the number of Segment Stores in the race. The high bound for this number of attempts is the total number of Segment Store instances in the cluster."

Reply:
Fixed.